controller/array_action/array_mediator_svc.py

```diff
@@ -194,12 +194,13 @@ def _generate_volume_response(self, cli_volume):
             source_volume_wwn,
             self.array_type)
 
-    def _generate_snapshot_response(self, cli_snapshot, source_volume_name):
+    def _generate_snapshot_response(self, cli_snapshot):
+        source_volume_wwn = self._get_source_volume_wwn_if_exists(cli_snapshot)
```

Comment:
since `create_snapshot` already has `source_cli_volume`, and `_generate_snapshot_response_with_verification` already has `fcmap` (as target), I think `_generate_snapshot_response` should require `source_volume_id` (as in ds8k), even though it would require `_generate_snapshot_response_with_verification` to call `_get_wwn_by_volume_name_if_exists`

Reply:
you just said the opposite! to do it like in `_generate_volume_response` in `svc`

Reply:
this inconsistency is confusing - is there a reason for them to be different? and if not, which of them do you think is better?

Reply:
cause in ds8k no need to call anything for the value of volume_id.

Reply:
in ds8k indeed there is no need for the call, so IMO it's a good reason for the mediators to be different.
in svc, is there a reason for `_get_source_volume_wwn_if_exists` to be called, even though `create_snapshot` already has the `source_cli_volume`?

Reply:
because `create_snapshot`  is not the only one that calls it, `get_object_by_id` calls it too and don't have the source_volume id

Reply:
done